client/landing/stepper/declarative-flow/hosted-site-migration-flow.ts

```diff
@@ -1,9 +1,13 @@
+import { isEnabled } from '@automattic/calypso-config';
 import { HOSTED_SITE_MIGRATION_FLOW } from '@automattic/onboarding';
 import { type Flow } from './internals/types';
+import migration from './migration';
 import siteMigration from './site-migration-flow';
 
+const baseFlow = isEnabled( 'migration-flow/revamp' ) ? migration : siteMigration;
+
 const hostedSiteMigrationFlow: Flow = {
```

Comment:
It will break the current hosted-site migration flow when we enable it.

Reply:
Maybe we can add the feature flag directly on https://github.com/Automattic/wp-calypso/blob/7749853c5bfa4d8df361547d06fb95fde67273d3/client/landing/stepper/declarative-flow/registered-flows.ts#L173-L178

Reply:
> It will break the current hosted-site migration flow when we enable it.

I tested it, and it worked properly. Maybe I'm missing a scenario?

Also, I think we'd have a cons with this approach. The `variantSlug` would be different, so in the middle of the flow it would redirect to the other flow URL  (`/migration/`). Probably not a big issue, but if we could avoid it it would be better.

I also tested adding the feature flag directly to the `registered-flow.ts` as you suggested, but it didn't work because the flow name in the flow object is different. I got a white screen. Anyway, if we really need this change, maybe we could add another feature flag check inside the object to change the flow name.

Reply:
> I tested it, and it worked properly. Maybe I'm missing a scenario?

I could see the tests failing.

I enabled the feature flag on test.json and ran the `hosted-site-migration-flow.ts`

<img width="696" alt="image" src="https://github.com/user-attachments/assets/5abb0510-410a-45ad-a72c-9bd64d242d56">

Reply:
Maybe we can add a use effect to redirect the user from the hosted-site-migration to the new migration using all query params.

It will help us during the A/B test.

Reply:
I think the `useEffect` you referred to would be in the first step of the flow, right? I think I found a cleaner way to keep the tests running for this flow, while activating the feature flag of the new flow. WDYT? https://github.com/Automattic/wp-calypso/pull/93417/commits/25cfe69af12022b121366d79652f138cec5fe7be

Reply:
> I think the useEffect you referred to would be in the first step of the flow, right?
Sorry, I described the wrong hooke name, the correct one is the flow > useSideEffect.Ideally we should not have flow-specific logic on the steps.


I have mixed feelings regarding the current approach because we will not have room to handle any specific scenario if needed. 

Example:
`hosted-site-migration` receives a param X to cover a specific edge case and we are not handling it on the migration flow. 

I don't know if we are going to have it. if we have we need to jump to another approach. 

BUT, I currently do not have a practical case at the moment.

What do you think?

Reply:
Thinking a bit better. 

I think the current solution will not work because they are not compatible. 

E.g  All `/move` CTAs are pointing directly to `/sitePicker` instead of the migration root, so a custom mapping probably will be required.

Reply:
Hey @gabrielcaires! I tweaked the PR with your suggestion and updated the PR description.
Notice that it's not converting 2 scenarios yet. I'll open issues for them.